src/firefly/js/charts/ui/Histogram.jsx

```diff
@@ -63,39 +67,38 @@ export class Histogram extends React.Component {
     }
 
     shouldComponentUpdate(nextProps) {
-        const {series, data, width, height, logs, reversed, desc, binColor} = this.props;
-        // should rerender only if data or bin color has changed
+        const {series, data, width, height, logs, xAxis, yAxis, desc, binColor} = this.props;
+        // should rerender if data, logs, or bin color has changed
         // otherwise just change the existing chart
-        if (series !== nextProps.series || data !== nextProps.data || binColor !== nextProps.binColor) { return true; }
+        if (this.error || series !== nextProps.series || data !== nextProps.data || logs !== nextProps.logs || binColor !== nextProps.binColor) { return true; }
         const chart = this.refs.chart && this.refs.chart.getChart();
         if (chart) {
-            let doUpdate = false;
-            if (height !== nextProps.height || width !== nextProps.width ) {
-                chart.setSize(nextProps.width, nextProps.height, false);
-                return false;
-            }
+            try {
+                let doUpdate = false;
+                if (height !== nextProps.height || width !== nextProps.width ) {
+                    chart.setSize(nextProps.width, nextProps.height, false);
+                    return false;
```

Comment:
this function is a bit confusing to me.  beside returning should update, it also does a chart.redraw.
but here, a setSize() does not need to be redrawn?